Homework3/Homework3/DuplicateUserCredentialsException.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+class DuplicateUserCredentialsException : Exception
+{
+    public DuplicateUserCredentialsException()
+    {
+    }
+
+    public DuplicateUserCredentialsException(string message) : base(message)
```

Comment:
Instead, maybe you can say which user is duplicate and just pass a username rather than the full message? You could internally form the message as you want